webapp/tests/mocha/unit/enketo/medic-xpath-extensions.spec.js

```diff
@@ -175,7 +175,7 @@ describe('medic-xpath-extensions', function () {
     const diffInDays = func['cht:difference-in-days'];
 
     it('should handle cases where end date is before start date and return negative number', function () {
-      const result = diffInDays(wrapDate('2023-08-01'), wrapDate('2023-01-01'));
+      const result = diffInDays(wrapDate('2023-08-01 UTC'), wrapDate('2023-01-01 UTC'));
```

Comment:
This breaks my brain. Does it really make sense for dates to have timezones? Or should this be called a datezone? I don't think this is sensible.

Secondly the test should reflect what the field value might actually have in an xform. This may be a good edge case test but the common case will be a date without a timezone so we need to decide what we're doing to do with that (ie: the original string).

If the function is being given the same date input and coming up with a different answer depending on which timezone you're in then there's a bug in the function. The way this might manifest in real life is one user may see a valid result and another would see an invalid one. This would make testing configurations difficult.

I think a better solution would be to revert this change and modify the code so for dates it works consistently regardless of timezones.

Reply:
FTR [this change](https://github.com/medic/cht-core/pull/9017/files#r1571480910) fixed the issues here so that these changes were not necessary.